src/loki/search.cc

```diff
@@ -580,17 +581,22 @@ struct bin_handler_t {
         bool reachable = reach.outbound >= p_itr->location.min_outbound_reach_ &&
                          reach.inbound >= p_itr->location.min_inbound_reach_;
         // it's possible that it isnt reachable but the opposing is, switch to that if so
-        const GraphTile* opp_tile = tile;
-        const DirectedEdge* opp_edge = nullptr;
-        if (!reachable && (opp_edge = reader.GetOpposingEdge(edge, opp_tile)) &&
-            edge_filter(opp_edge) > 0.f) {
-          auto opp_reach = check_reachability(begin, end, opp_tile, opp_edge, edge_id);
-          if (opp_reach.outbound >= p_itr->location.min_outbound_reach_ &&
-              opp_reach.inbound >= p_itr->location.min_inbound_reach_) {
-            tile = opp_tile;
-            edge = opp_edge;
-            reach = opp_reach;
-            reachable = true;
+        if (!reachable) {
+          const GraphTile* opp_tile = tile;
+          const GraphId opp_edgeid = reader.GetOpposingEdgeId(edge_id, opp_tile);
+          if (!opp_edgeid)
```

Comment:
similar nesting addition as the comment i made above. maybe there is a better way to get both the opposing edge and id in one call so we can avoid the nesting